analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -60,7 +60,7 @@ class Gradle : PackageManager() {
     companion object : PackageManagerFactory<Gradle>(
             "https://gradle.org/",
             "Java",
-            listOf("build.gradle")
+            listOf("[a-zA-Z0-9]*.gradle")
```

Comment:
I've never seen non-`build.gradle` Gradle build files in real life. Do you have a concrete example for this?

Reply:
Some of Gradle subprojects use `subproject_name.gradle` names: 
https://github.com/gradle/gradle/tree/master/subprojects/wrapper

Reply:
@pnowakowski good finding, I suggest that we combine it with Sebastians approach to SBT:

1) Files that are supposed to be included by gradle usually also use the ".gradle" extension, but we don't want to match them (see https://docs.gradle.org/current/userguide/plugins.html#sec:script_plugins).
2) You are right, that we won't match e.g. the wrapper.gradle, but if you look at the settings.gradle of the root directory there is the mapping that defines the filenames. So in theory the files could be named anyhow.

-> As Sebastian is modifying the list of definition files for SBT in the prepareResolution() method I think we should do the same for Gradle, and ask Gradle for a list of subproject build files for each found build.gradle.

Reply:
The downside of this is if we only pass the wrapper directory to the analyzer it won't find any buildfiles.

@sschuberth & @pnowakowski: What do you two think?

Reply:
> [...] ask Gradle for a list of subproject build files for each found build.gradle.

I agree that would be the best way to go.

> The downside of this is if we only pass the wrapper directory to the analyzer it won't find any buildfiles.

That's acceptable, I believe. We'll hardly run into the case of scanning only the directory of a Gradle sub-project.

Reply:
I think we also need to check top level build gradle file name as it can be renamed as well.
`settings.gradle` should then have a line similar to `rootProject.buildFileName = 'some_name.some_ext'`

Reply:
Actually, if we agree to not be able to scan sub-project folders on their own, shouldn't it be enough to look for `build.gradle` *or* `settings.gradle`? Because if `build.gradle` was renamed on a root level, there must be a `settings.gradle` file to do so.

Reply:
That's true, because we start Gradle on a directory not a specific file it should be smart enough to figure out the actual build file name on its own.

Reply:
I wasn't sure how gradle dependency connector is executed exactly, so I checked and it seems you're right, that will be enough.

Reply:
Great, thanks for double-checking. Could you please update this PR the commit message accordingly? In the commit message, you may also drop the `.kt` extension from `Gradle.kt` and just say `Gradle: ...`.